### ISSUE TITLE
Check the text color constrast

### ISSUE THREAD

**Issue by palemieux** [w3c/imsc#391]
See https://github.com/w3c/imsc/pull/388#discussion_r195347983

@nigelmegitt to specify tool/criteria.

**Comment by nigelmegitt** [comment]
The [BBC's Mobile Accessibility Guidelines](http://www.bbc.co.uk/guidelines/futuremedia/accessibility/mobile/design/colour-contrast#TESTING) suggest 3 tools:

* [Snook.ca colour contrast check](http://snook.ca/technical/colour_contrast/colour.html), 
* [Webaim color contrast checker ](http://webaim.org/resources/contrastchecker/)or 
* [TPG colour contrast analyser](https://www.paciellogroup.com/resources/contrastanalyser/)

**Comment by nigelmegitt** [comment]
Using "normal" text for WebAIM:

Foreground|Background|snook|WebAIM|
----------|----------|-----|------|
green = #008000|white|[5.137:1 AA pass, AAA fail](https://snook.ca/technical/colour_contrast/colour.html#fg=008000,bg=FFFFFF)|[5.14:1 AA pass, AAA fail](https://webaim.org/resources/contrastchecker/?fcolor=008000&bcolor=FFFFFF)|
green = #008000|#f0f6ff|[4.729:1 AA pass, AAA fail](https://snook.ca/technical/colour_contrast/colour.html#fg=008000,bg=F0F6FF)|[4.73:1 AA pass, AAA fail](https://webaim.org/resources/contrastchecker/?fcolor=008000&bcolor=F0F6FF)|
#ff6666|white|[2.859:1 fail](https://snook.ca/technical/colour_contrast/colour.html#fg=FF6666,bg=FFFFFF)|[2.86:1 fail](https://webaim.org/resources/contrastchecker/?fcolor=FF6666&bcolor=FFFFFF)|
#ff6666|#f0f6ff|[2.632:1 fail](https://snook.ca/technical/colour_contrast/colour.html#fg=FF6666,bg=F0F6FF)|[2.63:1 fail](https://webaim.org/resources/contrastchecker/?fcolor=FF6666&bcolor=F0F6FF)|
orange = #ffa500|white|[1.974:1 fail](https://snook.ca/technical/colour_contrast/colour.html#fg=FFA500,bg=FFFFFF)|[1.97:1 fail](https://webaim.org/resources/contrastchecker/?fcolor=FFA500&bcolor=FFFFFF)|
orange = #ffa500|#f0f6ff|[1.817:1 fail](https://snook.ca/technical/colour_contrast/colour.html#fg=FFA500,bg=F0F6FF)|[1.82:1 fail](https://webaim.org/resources/contrastchecker/?fcolor=FFA500&bcolor=F0F6FF)|

Looks like we need better colours!

**Comment by nigelmegitt** [comment]
Playing with the colour picker in snook:

* choosing `#005F00` for green would work;
* choosing `#9F1B1B` for red would work;
* choosing `#734B00` for orange would kind-of work but doesn't look anything like orange.

[aside: I'm loving GitHub Markdown's blob of colour after those colour values!]

Another thing we could play with would be to put a background colour behind the text and use black or white foreground colour to get a better result.

For example:
* using `#006600` for background green and white foreground passes 
* using `#FF6666` for background red and black foreground passes
* using `#FFA500` for background orange and black foreground passes

Codepen that does that, by way of example: https://codepen.io/nigelmegitt/pen/dKVObZ

**Comment by spoeschel** [comment]
> - choosing #005F00 for green would work;
> - choosing #9F1B1B for red would work;
> - choosing #734B00 for orange would kind-of work but doesn't look anything like orange.

For me having red-green colour blindness, these colours are unfortunately not easily distinguishable.

I would prefer the background colour approach. The example you propose seems to be good. I would just prefer colours that are more close to real green or red, if possible. Does using different background (instead of text) colours maybe relax the contrast requirements? Would it possibly allow to use the current text colors of #388 as background colours?

**Comment by nigelmegitt** [comment]
@spoeschel The only colour that is different from the current text colours of #388 is the green one. 

Using `white` text on a `green` background fails WCAG AAA, though it passes AA. Adjusting just the "value" slider on the snook.ca viewer, starting from `#008000` and going darker, the first value that passes is `#006600`. Can you find a better colour that passes?

If you want the red to be redder, `#B60000` would pass as a background colour with white foreground.

**Comment by nigelmegitt** [comment]
A codepen showing those colours as per https://github.com/w3c/imsc/issues/391#issuecomment-397627563 but with a slightly paler orange of `#FFC000` is at https://codepen.io/nigelmegitt/pen/NzajYW - does that look better @spoeschel ?

**Comment by nigelmegitt** [comment]
Following @spoeschel 's comments about red and green being hard to tell apart for some people, I noticed that the words are also a similar shape, beginning and ending with the same letter and being about the same width on the screen, so I've updated the [codepen](https://codepen.io/nigelmegitt/pen/NzajYW) to include a check mark or a cross or an exclamation mark to make it even quicker and easier to tell them apart.

**Comment by palemieux** [comment]
I like the idea of adding a mark. Adding a background to the term will require the removal of the background on alternate rows, which is probably ok.

**Comment by nigelmegitt** [comment]
> Adding a background to the term will require the removal of the background on alternate rows

@palemieux I don't follow why?

I see that the current approach in the `issues-359-363-374-375-376-380-reformat-features-table+misc-editorial-issues` branch is to use `<td class="permitted">permitted ...` but surely a search and replace with `<td><span class="permitted label">permitted</span>` would be straightforward? (assuming you used the same styles as I defined in the codepen - obviously you could structure them a different way). Similarly for prohibited and permitted-deprecated, obviously.

That would have the small advantage of not needing to make the `td.permitted::first-line` have a special style, which is brittle in the case that any other text is ever placed on the same line as the word "permitted".

**Comment by spoeschel** [comment]
@nigelmegitt

>  I noticed that the words are also a similar shape, beginning and ending with the same letter and being about the same width on the screen

This is one of the reasons why it is difficult to distinguish the different dispositions and why we need additional means. Your idea with the marks provides a good additional indication (especially when the table is printed on a B/W printer) and should be adopted as well.

> does that look better @spoeschel ?

Regarding the prohibited and permitted colours, comparing both next to each other is quite OK, but the colour combination is not suitable for just catching a glimpse unfortunately.

With white foreground colour I wasn't able to find a suitable combination with snook.ca that passes WCAG 2 AAA Compliance, but with black foreground for all three dispositions, it seems to work:

https://codepen.io/anon/pen/XYeEoy
- #50FF50 for permitted
- #FF5F5F for prohibited
- #FFC000 for permitted-deprecated

[I wasn't able to find out how you added the colour boxes]

**Comment by nigelmegitt** [comment]
@spoeschel the trick is to enclose the hex RGB value in backticks like so: 

* `#50FF50` for permitted
*  `#FF5F5F` for prohibited
*  `#FFC000` for permitted-deprecated

**Comment by nigelmegitt** [comment]
@spoeschel the colours you have in https://github.com/w3c/imsc/issues/391#issuecomment-397728217 look fine to me. I'm happy with that option also.
